Warning critical dependency

First of all James thanks for publishing this cool project 🙌 , i have tried and its working really good in SSR too, but i'm getting this warning in console and i don't know yet why! if anyone else is getting please let me know and if you did get rid of it.


```
WARNING in ./~/react-loadable/lib/index.js
31:24-41 Critical dependency: the request of a dependency is an expression
```

FWIW, I verified that this is nothing more than an annoyance. All it adds to your bundle is this:

```javascript
/* 1 */
/***/ (function(module, exports) {

function webpackEmptyContext(req) {
	throw new Error("Cannot find module '" + req + "'.");
}
webpackEmptyContext.keys = function() { return []; };
webpackEmptyContext.resolve = webpackEmptyContext;
module.exports = webpackEmptyContext;
webpackEmptyContext.id = 1;


/***/ }),
```

--- a/flow-typed/webpack.js
+++ b/flow-typed/webpack.js
@@ -1,0 +1,2 @@
+declare function __webpack_require__(pathOrId: string | number): any;
+declare var __webpack_modules__: {};

--- a/src/index.js
+++ b/src/index.js
@@ -5,6 +5,9 @@ type GenericComponent<Props> = Class<React.Component<{}, Props, mixed>>;
 type LoadedComponent<Props> = GenericComponent<Props>;
 type LoadingComponent = GenericComponent<{}>;
 
+let isWebpack = typeof __webpack_require__ !== "undefined";
+let requireFn = isWebpack ? __webpack_require__ : module.require.bind(module);
+
 let babelInterop = obj => {
   // $FlowIgnore
   return obj && obj.__esModule ? obj.default : obj;
@@ -13,13 +16,10 @@ let babelInterop = obj => {
 let tryRequire = (pathOrId: string | number) => {
   try {
     // $FlowIgnore
-    return babelInterop(require(pathOrId));
+    return babelInterop(requireFn(pathOrId));
   } catch (err) {}
   return null;
 };
-
-// $FlowIgnore
-let isWebpack = typeof __webpack_modules__ !== "undefined";
 
 type Options = {
   loader: () => Promise<LoadedComponent<Props>>,
@@ -42,12 +42,19 @@ export default function Loadable<Props: {}, Err: Error>(opts: Options) {
   let outsidePromise = null;
   let outsideError = null;
 
-  if (serverSideRequirePath) {
+  if (!isWebpack && serverSideRequirePath) {
     outsideComponent = tryRequire(serverSideRequirePath);
   }
 
   if (isWebpack && webpackRequireWeakId) {
-    outsideComponent = tryRequire(webpackRequireWeakId());
+    let weakId = webpackRequireWeakId();
+    if (__webpack_modules__[weakId]) {
+      // if it's not in webpack modules, we wont be able
+      // to load it. If we attempt to, we mess up webpack's
+      // internal state, so only tryRequire if it's already
+      // in webpack modules.
+      outsideComponent = tryRequire(weakId);
+    }
   }
 
   let load = () => {

